test/browser_tests/spec_suites/organisms/header.js

```diff
@@ -135,11 +134,31 @@ describe( 'Header', function() {
               expect( _dom.overlay.isDisplayed() ).toBe( true );
             } );
         } );
+
+        describe( 'then click search', function() {
+          it( 'should show the search and hide menu', function() {
+            browser.driver.actions().click( _dom.megaMenuTrigger ).perform();
+            browser.driver.wait( _dom.megaMenuContent.isDisplayed() );
+            browser.driver.actions().click( _dom.globalSearchTrigger ).perform()
+              .then( function() {
+                expect( _dom.globalSearch.isDisplayed() ).toBe( true );
+
+              } );
+          } );
+        } );
+      } );
+
+      describe( 'click search, then click mega menu', function() {
+        it( 'should show the megamenu and hide search', function() {
+          browser.driver.actions().click( _dom.globalSearchTrigger ).perform();
+          browser.driver.wait( _dom.megaMenuContent.isDisplayed() );
```

Comment:
This is what I was running into as well. `isDisplayed()` has cases where it will return `true` even when the element is offscreen and visible (in absolute positioned elements, and, it seems, with elements [that have tranform applied](https://github.com/seleniumhq/selenium-google-code-issue-archive/issues/8054#issuecomment-192200898)). See some background here: https://github.com/angular/protractor/issues/759

With the depressing conclusion:
> sometimes elements that you don't think are displayed are being reported as displayed. I think this is because WebDriver is not smart enough to cover all cases of elements that don't have display:false but are not on the screen for some reason. If you can find a reduced case showing one of these issues we can report to WebDriver, but otherwise there's nothing Protractor can do.